Vendored dependencies can become stale on platforms that use GNUMakefile

**Describe the bug**
If `vendor/` is already populated by repositories associated with the project's dependencies, the [`git clone` commands](https://github.com/plasma-umass/scalene/blob/cd953be1bf911cb25ca6e5e912acf02107edc6ad/GNUmakefile#L56-L67) in `GNUMakefile` won't end up updating these commands. In the best case, this will cause a build error, and in the worst case the build will succeed by use the wrong versions of these dependencies.

In my case, I had a pretty old copy of `vendor/Heap-Layers` from a previous build, and before I worked out what was going on, I was seeing build errors of the form:

```
src/source/libscalene.cpp: In static member function ‘static void* MakeLocalAllocator<Domain>::local_malloc(void*, size_t)’:                                          
src/source/libscalene.cpp:165:23: error: ‘register_malloc’ is not a member of ‘TheHeapWrapper’ {aka ‘HL::HeapWrapper<CustomHeapType, 8388608>’}                       
  165 |       TheHeapWrapper::register_malloc(len, ScaleneHeader::getObject(header));                                                                                 
      |                       ^~~~~~~~~~~~~~~                                                                                                                         
src/source/libscalene.cpp: In static member function ‘static void MakeLocalAllocator<Domain>::local_free(void*, void*)’:                                              
src/source/libscalene.cpp:192:25: error: ‘register_free’ is not a member of ‘TheHeapWrapper’ {aka ‘HL::HeapWrapper<CustomHeapType, 8388608>’}                         
  192 |         TheHeapWrapper::register_free(sz, ptr);                                                                                                               
      |                         ^~~~~~~~~~~~~                                      
src/source/libscalene.cpp: In static member function ‘static void* MakeLocalAllocator<Domain>::local_realloc(void*, void*, size_t)’:                                  
src/source/libscalene.cpp:214:25: error: ‘register_malloc’ is not a member of ‘TheHeapWrapper’ {aka ‘HL::HeapWrapper<CustomHeapType, 8388608>’}                       
  214 |         TheHeapWrapper::register_malloc(new_size - sz,                                                                                                        
      |                         ^~~~~~~~~~~~~~~                                                                                                                       
src/source/libscalene.cpp:217:25: error: ‘register_free’ is not a member of ‘TheHeapWrapper’ {aka ‘HL::HeapWrapper<CustomHeapType, 8388608>’}
  217 |         TheHeapWrapper::register_free(sz - new_size, ptr);
      |                         ^~~~~~~~~~~~~
In file included from src/source/libscalene.cpp:18:
src/include/sampleheap.hpp: In instantiation of ‘class SampleHeap<1, HL::NullHeap<MakeLocalAllocator<Domain>::local_malloc(void*, size_t) [with PyMemAllocatorDomain D
omain = PYMEM_DOMAIN_MEM; size_t = long unsigned int]::Nada> >’:
src/source/libscalene.cpp:169:52:   required from ‘static void* MakeLocalAllocator<Domain>::local_malloc(void*, size_t) [with PyMemAllocatorDomain Domain = PYMEM_DOMA
IN_MEM; size_t = long unsigned int]’
src/source/libscalene.cpp:113:16:   required from ‘MakeLocalAllocator<Domain>::MakeLocalAllocator() [with PyMemAllocatorDomain Domain = PYMEM_DOMAIN_MEM]’
src/source/libscalene.cpp:242:45:   required from here
src/include/sampleheap.hpp:56:8: error: ‘Alignment’ is not a member of ‘HL::NullHeap<MakeLocalAllocator<Domain>::local_malloc(void*, size_t) [with PyMemAllocatorDomai
n Domain = PYMEM_DOMAIN_MEM; size_t = long unsigned int]::Nada>’
   56 |   enum { Alignment = SuperHeap::Alignment };
      |        ^
src/include/sampleheap.hpp: In instantiation of ‘class SampleHeap<1, HL::NullHeap<MakeLocalAllocator<Domain>::local_malloc(void*, size_t) [with PyMemAllocatorDomain D
omain = PYMEM_DOMAIN_OBJ; size_t = long unsigned int]::Nada> >’:
src/source/libscalene.cpp:169:52:   required from ‘static void* MakeLocalAllocator<Domain>::local_malloc(void*, size_t) [with PyMemAllocatorDomain Domain = PYMEM_DOMA
IN_OBJ; size_t = long unsigned int]’
src/source/libscalene.cpp:113:16:   required from ‘MakeLocalAllocator<Domain>::MakeLocalAllocator() [with PyMemAllocatorDomain Domain = PYMEM_DOMAIN_OBJ]’
src/source/libscalene.cpp:243:45:   required from here
src/include/sampleheap.hpp:56:8: error: ‘Alignment’ is not a member of ‘HL::NullHeap<MakeLocalAllocator<Domain>::local_malloc(void*, size_t) [with PyMemAllocatorDomai
n Domain = PYMEM_DOMAIN_OBJ; size_t = long unsigned int]::Nada>’
In file included from vendor/Heap-Layers/heaps/./top/all.h:4,
                 from vendor/Heap-Layers/heaps/all.h:8,
                 from vendor/Heap-Layers/heaplayers.h:97,
                 from src/source/libscalene.cpp:8:

// and so on…
```

**To Reproduce**
Assuming that `vendor/Heap-Layers` already exists and contains the associated git repo, the following commands should reproduce the error I was seeing:

```
$ git checkout 48ad8a4  # current commit on branch master
…
$ git -C vendor/Heap-Layers checkout dae2485
…
$ make bdist
```

**Expected behavior**
The `GNUMakefile` should have and use a target to clear the vendor targets like [`Makefile` does](https://github.com/plasma-umass/scalene/blob/cd953be1bf911cb25ca6e5e912acf02107edc6ad/Makefile#L41-L45)

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: Ubuntu 20.04.3
 - Version many (see above)

**Additional context**
Discovered while looking into #503 and trying to verify that I could build the latest version without any of the changes associated with that issue

--- a/GNUmakefile
+++ b/GNUmakefile
@@ -61,7 +61,10 @@ vendor/printf/printf.cpp:
 	cd vendor/printf && ln -s printf.c printf.cpp
 	sed $(SED_INPLACE) -e 's/^#define printf printf_/\/\/&/' vendor/printf/printf.h
 
-vendor-deps: vendor/Heap-Layers vendor/printf/printf.cpp
+clear-vendor-dirs:
+	rm -fr vendor/
+
+vendor-deps: clear-vendor-dirs vendor/Heap-Layers vendor/printf/printf.cpp
 
 mypy:
 	-mypy $(PYTHON_SOURCES)

